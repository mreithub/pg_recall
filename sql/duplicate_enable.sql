--
-- Creates a simple table and enables logging twice (the second call should update the interval)
--

BEGIN;

CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT NOT NULL
);

-- enable logging
SELECT recall.enable('config', '2 months');

-- run a cleanup to make sure last_cleanup is set
SELECT recall.cleanup_all();

-- check the config table
SELECT tblid, log_interval, now() - last_cleanup FROM recall._config;


-- a new call to recall.enable should trigger a notice, update the interval
-- and reset last_cleanup
SELECT recall.enable('config', '3 months');
SELECT tblid, log_interval, now() - last_cleanup FROM recall._config;

ROLLBACK;
