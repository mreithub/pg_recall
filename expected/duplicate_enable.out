--
-- Creates a simple table and enables logging twice (the second call should update the interval)
--
BEGIN;
CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT NOT NULL
);
-- enable logging
SELECT recall.enable('config', '2 months');
 enable 
--------
 
(1 row)

-- run a cleanup to make sure last_cleanup is set
SELECT recall.cleanup_all();
NOTICE:  recall: Cleaning up table config
CONTEXT:  SQL statement "SELECT recall.cleanup(tbl)"
PL/pgSQL function recall.cleanup_all() line 7 at PERFORM
 cleanup_all 
-------------
 
(1 row)

-- check the config table
SELECT tblid, log_interval, now() - last_cleanup FROM recall._config;
 tblid  | log_interval | ?column? 
--------+--------------+----------
 config | @ 2 mons     | @ 0
(1 row)

-- a new call to recall.enable should trigger a notice, update the interval
-- and reset last_cleanup
SELECT recall.enable('config', '3 months');
NOTICE:  recall.enable(config, @ 3 mons) called on an already managed table. Updating log_interval and pkey_cols, clearing last_cleanup
 enable 
--------
 
(1 row)

SELECT tblid, log_interval, now() - last_cleanup FROM recall._config;
 tblid  | log_interval | ?column? 
--------+--------------+----------
 config | @ 3 mons     | 
(1 row)

ROLLBACK;
